app/livechat/imports/server/rest/sms.js

```diff
@@ -77,30 +86,57 @@ API.v1.addRoute('livechat/sms-incoming/:service', {
 			},
 		};
 
-		sendMessage.message.attachments = sms.media.map((curr) => {
+		let file;
+		const attachments = Promise.await(Promise.all(sms.media.map(async (curr) => {
+			const { url: twilioUrl, contentType } = curr;
+
+			if (!file) {
```

Comment:
I didn't understand why you're looping all entries from `sms.media` but is only saving a single file. is there a reason for that? with proposed changes a message is being created with multiple `attachments` but all of them referencing the same file, is that the expected behavior?

Reply:
@sampaiodiego,  I just fixed it. In fact, due to the way the relationship between the message and the upload is [currently designed (1 - 1)](https://github.com/RocketChat/Rocket.Chat.Apps-engine/blob/alpha/src/definition/messages/IMessage.ts#L21), there is no reason to store all attachments in the array, except the first. Otherwise, the files would be stored, but through the message, only one file can be linked. Once (if) we redesign the association between messages and uploads, we can store more than one upload per message.